contracts/root/depositManager/DepositHelper.sol

```diff
@@ -0,0 +1,52 @@
+pragma solidity ^0.5.2;
+
+import { IERC20 } from "openzeppelin-solidity/contracts/token/ERC20/IERC20.sol";
+import { IERC721 } from "openzeppelin-solidity/contracts/token/ERC721/IERC721.sol";
+
+import { Registry } from "../../common/Registry.sol";
+import { IDepositManager } from "./IDepositManager.sol";
+
+contract DepositHelper {
+
+  Registry private registry;
+
+  enum Type { Invalid, ERC20, ERC721 }
+  mapping(address => Type) public tokenType;
```

Comment:
Yeah, but I wanted to avoid calling an external contract for every transfer! But I guess there is no other way because even this cache will be outdated if we un-map a token.